exercises/practice/word-search/Tests/WordSearchTests/WordSearchTests.swift

```diff
@@ -0,0 +1,542 @@
+      "screeaumgr", "alxhpburyi", "jalaycalmp", "clojurermt",
+    ]
+    let words = [
+      "clojure", "elixir", "ecmascript", "rust", "java", "lua", "lisp", "ruby", "haskell",
+    ]
+
+    var expected = [String: WordLocation?]()
+    expected["clojure"] = WordLocation(
+      start: ["column": 1, "row": 10],
+      end: ["column": 7, "row": 10]
+    )
+    expected["ecmascript"] = WordLocation(
+      start: ["column": 10, "row": 1],
+      end: ["column": 10, "row": 10]
+    )
+    expected["elixir"] = WordLocation(
+      start: ["column": 6, "row": 5],
+      end: ["column": 1, "row": 5]
+    )
+    expected["haskell"] = nil
+    expected["java"] = WordLocation(
+      start: ["column": 1, "row": 1],
+      end: ["column": 4, "row": 4]
+    )
+    expected["lisp"] = WordLocation(
+      start: ["column": 3, "row": 6],
+      end: ["column": 6, "row": 3]
+    )
+    expected["lua"] = WordLocation(
+      start: ["column": 8, "row": 9],
+      end: ["column": 6, "row": 7]
+    )
+    expected["ruby"] = WordLocation(
+      start: ["column": 8, "row": 6],
+      end: ["column": 5, "row": 9]
+    )
+    expected["rust"] = WordLocation(
+      start: ["column": 9, "row": 5],
+      end: ["column": 9, "row": 2]
+    )
+
+    #expect(search(words: words, in: grid) == expected)
+  }
+
+  @Test(
+    "Should fail to locate words that are not on horizontal, vertical, or diagonal lines",
+    .enabled(if: RUNALL))
+  func testShouldFailToLocateWordsThatAreNotOnHorizontalVerticalOrDiagonalLines() {
+    let grid = ["abc", "def"]
+    let words = ["aef", "ced", "abf", "cbd"]
+
+    var expected = [String: WordLocation?]()
+    expected["abf"] = nil
+    expected["aef"] = nil
+    expected["cbd"] = nil
+    expected["ced"] = nil
+
+    #expect(search(words: words, in: grid) == expected)
+  }
+
+  @Test("Should not concatenate different lines to find a horizontal word", .enabled(if: RUNALL))
+  func testShouldNotConcatenateDifferentLinesToFindAHorizontalWord() {
+    let grid = ["abceli", "xirdfg"]
+    let words = ["elixir"]
+
+    var expected = [String: WordLocation?]()
+    expected["elixir"] = nil
+
+    #expect(search(words: words, in: grid) == expected)
+  }
+
+  @Test("Should not wrap around horizontally to find a word", .enabled(if: RUNALL))
+  func testShouldNotWrapAroundHorizontallyToFindAWord() {
+    let grid = ["silabcdefp"]
+    let words = ["lisp"]
+
+    var expected = [String: WordLocation?]()
+    expected["lisp"] = nil
+
+    #expect(search(words: words, in: grid) == expected)
+  }
+
+  @Test("Should not wrap around vertically to find a word", .enabled(if: RUNALL))
+  func testShouldNotWrapAroundVerticallyToFindAWord() {
+    let grid = ["s", "u", "r", "a", "b", "c", "t"]
+    let words = ["rust"]
+
+    var expected = [String: WordLocation?]()
+    expected["rust"] = nil
+
+    #expect(search(words: words, in: grid) == expected)
+  }
+}
+
+extension WordLocation {
```

Comment:
I get the idea that putting this in the source file would clutter it quite a bit. 

I think a better approach would be to put this whole struct as a whole in a separate source file. As you can see being done here: https://github.com/exercism/swift/tree/main/exercises/practice/saddle-points/Sources/SaddlePoints.